src/sdk/object_io.js

```diff
@@ -495,6 +498,7 @@ class ObjectIO {
             });
             await mc.run();
             if (mc.had_errors) throw new Error('Upload map errors');
+            if (params.async_update_cache_stats()) params.async_update_cache_stats(params.range.end - params.range.start);
```

Comment:
we don't need it to be called async, and the condition should not run the function - just check for it's existence.
```suggestion
            if (params.update_cache_stats) params.update_cache_stats(params.range.end - params.range.start);
```

Reply:
I suggested to call this property `upload_chunk_hook` - what do you think?